lib/inspec/resources/ssh_key.rb

```diff
@@ -0,0 +1,123 @@
+require "inspec/utils/file_reader"
+require "net/ssh" unless defined?(Net::SSH)
+
+# Change module if required
+module Inspec::Resources
+  class SshKey < FileResource
+    # Every resource requires an internal name.
+    name "ssh_key"
+
+    # Restrict to only run on the below platforms (if none were given,
+    # all OS's and cloud API's supported)
+    supports platform: "unix"
+    supports platform: "windows"
+
+    desc "public/private SSH key pair test"
+
+    example <<~EXAMPLE
+      describe ssh_key('~/.ssh/id_rsa') do
+        its('key_length') { should eq 4096 }
+        its('type') { should cmp /rsa/ }
+        it { should be_private }
+      end
+    EXAMPLE
+
+    include FileReader
+
+    def initialize(keypath, passphrase = nil)
+      skip_resource "The `ssh_key` resource is not yet available on your OS." unless inspec.os.unix? || inspec.os.windows?
+      @key_path = keypath
+      @passphrase = passphrase
+      @key = read_ssh_key(read_file_content(@key_path, allow_empty: true), passphrase)
```

Comment:
Why are we sending `allow_empty` as `true` ? Is it because the ssh key file can be empty and we don't want to raise any error?

Reply:
Good catch. Let me verify this.